### ISSUE TITLE
[1.16.5-7.4.0] Fluid routers not activating properly

### ISSUE THREAD

**Issue by CodeGordon** [desht/ModularRouters#136]
### Minecraft Version
1.16.5

### Forge Version
36.1.32

### Mod Version
7.4.0

### Describe your problem, including steps to reproduce it
Fluid routers not working, or working only intermittently. I have 9 speed upgrades and 35 fluid upgrades on four routers, using a pair of Fluid Mk2 modules whitelisting the Blood Magic life essence fluid, and buckets in each router's buffer slot. Three are pulling from a Blood Magic altar and inserting into a Mekanism tank, and one doing the reverse elsewhere on a separate altar and tank. I have all four routers set to operate on a high signal, and an Integrated Dynamics system with redstone writers to activate them. Other routers work just fine, but the fluid transfer doesn't work very well. I have adjusted each fluid module's transfer rate to match the transfer rate on their router (800 mB/tick with the above upgrades), but that doesn't change their behavior. Setting routers to Always instead of High Signal doesn't make them work either, so they appear to just not be functioning at all.

### Any other comments?
Attaching log file in case that's helpful.
[latest.log](https://github.com/desht/ModularRouters/files/6836497/latest.log)

**Comment by desht** [comment]
IIRC Blood Magic altars have a very low fluid transfer rate by default - runes of displacement (I think) are needed to improve that.  That's probably your issue here.  Especially if you're seeing intermittent functioning - sounds like the altar just isn't allowing you to pull that fast.

**Comment by CodeGordon** [comment]
In the several hours since posting this issue, there hasn't been any movement from any of the four routers into or out of either altar. The tank near the receiving altar has the same 2000 mB in it that was present earlier when I manually added some from a bucket during my tests.

I hadn't considered the transfer rate built-in to the altars, but I added 18 Acceleration and 10 Displacement runes to the sending altar, with no change. Nothing is transferring through any of the four routers in any way. I've double checked the Integrated Dynamics system and the redstone writers are properly set to issue a strong signal, and the variable cards are all correct. The writers are receiving a true boolean, so they should be emitting the proper redstone signals, as this is a setup I've used multiple times before, even with other routers. The only difference with this setup is that these are moving fluids instead of items.

**Comment by desht** [comment]
Ah, wait - I just noticed: you're using buckets in the router?  That isn't going to work, unless you can pull 1000mB fluid at a time, which is very unlikely to possible with a blood altar.  Try putting some modded fluid container in the router instead: ones I know for sure will work are PneumaticCraft's tanks, and Immersive Engineering's Jerrycans (Mekanism tanks probably won't work here).

(Of course if you have PneumaticCraft in the pack, Liquid Hoppers are probably easier still...)

**Comment by CodeGordon** [comment]
I'll try that, then. As mentioned, I have Mekanism tanks to use, but I thought based on the way the in-game book is worded that you just needed a thing that can hold fluid as a placeholder, and the router behaves as it needs to behave. If that's the problem, then it should probably be more clearly stated that the buffer item needs to be able to hold however many mB you're trying to transfer at a time, and support being partially filled if necessary. So, buckets should only be suggested if moving whole buckets at once.

Incidentally, I was able to pipe out of the sending altar with Mekanism pipes and a quantum entangloporter, set up a new altar for my orb, and set up another quantum link to both the new one and the original receiving altar with more pipes. The pipes work fine, and with the exception of the original receiving altar, they're able to be hidden underneath.

**Comment by desht** [comment]
I thought it was fairly clear that a vanilla bucket can hold either 0 or 1000mB of fluid :P  That's basic vanilla mechanics.

**Comment by CodeGordon** [comment]
Yes, but the way it read made me think that the router has a separate fluid buffer that it uses with a fluid-holding item in the item buffer. Not that it's moving fluids into and out of the item in its buffer, but that it's handling fluids separately.

**Comment by desht** [comment]
OK, I can make that clearer.   It's possible future versions of the mod will have their own internal fluid tank (similar to how 7.5.0 adds an internal energy buffer) which will make this simpler.

**Comment by CodeGordon** [comment]
Just tried the Mekanism tank and that worked, and reread the in-game manual. It does say that the router doesn't have an internal tank. So I guess I'm just dumb. Lol.
